### PR TITLE
Islandora 1743 -- Label for simple search

### PR DESCRIPTION
**Pull request by dmcollins**
**JIRA Ticket**: [https://jira.duraspace.org/browse/ISLANDORA-1743](https://jira.duraspace.org/browse/ISLANDORA-1743)

* Other Relevant Links (Google Groups discussion, related pull requests, Release pull requests, etc.)

# What does this Pull Request do?

Addresses issue of a hardcoded label attached to the Simple Search input. Allows users to change label text and choose to visibly hide the label, while maintaining label availability for screen-readers, etc.

# What's new?
Introduces a new group of settings under Solr Settings for the Simple Search Block, constructs the input label using these settings.

screen shot 2017-05-26 at 15 58 37

# How should this be tested?

Once implemented, insure that label text and visibility respond to controls.

# Additional Notes:
Default settings are to use the same label text as currently hard-coded and defaults to visible.
Label text is required in order to meet accessibility guidelines.

# Interested parties
Tag (@ mention) interested parties or, if unsure, @Islandora/7-x-1-x-committers
Ping Maintainer Nelson Hart (@nhart)
Ping Melissa Anez (@manez) -- Interested party on JIRA ticket

### CODE REVIEW

File: includes/blocks.inc [hunk 1]
```diff
@@ -427,10 +427,21 @@ function islandora_solr_simple_search_form($form, &$form_state) {
       ),
     ),
   );
+
+  if (variable_get('islandora_solr_simple_search_label_visibility', 'TRUE') == FALSE) {
```
Comment (whikloj) [inline includes/blocks.inc]: Looks like your default should be `TRUE` not `'TRUE'`

**Comment by willtp87** [comment]
Should this be implemented in the block config?

**Comment by dmcollins** [comment]
@willtp87 Maybe? I put it here as nothing else in the module was configured at that level and I thought it made sense to put it near the settings for the advanced search block to make it easier for folks to find. Happy to move things around, though, if that's the preferred route.

edited to add: Presuming I understand you right and that you mean it should appear under admin/structure/block/manage/islandora_solr/simple/configure rather than admin/islandora/search/islandora_solr/settings -- I'm still relatively new to doing this so it occurred to me that you might have meant something else.

**Comment by whikloj** [comment]
@dmcollins I think because you might want someone with less privileges to be able to adjust the label, so putting it in the block gives you that freedom.

**Comment by willtp87** [comment]
I think it is also the most expected place for the configuration.  Indeed I suspect 
some of the existing configuration probably shouldn't live where it does, but hindsight is 20/20.

**Comment by dmcollins** [comment]
@whikloj @willtp87 This all makes sense! I'll get to work on rearranging things.

**Comment by dmcollins** [comment]
@whikloj @willtp87 This has been updated to move the functionality over to block config

File: includes/blocks.inc [hunk 2]
```diff
@@ -106,6 +106,40 @@ function islandora_solr_block_info() {
...
+ */
+function islandora_solr_block_configure($delta = '') {
+  if ($delta == 'simple') {
+    $form = array();
+    $form['islandora_solr_simple_search_label_title'] = array(
+      '#type' => 'textfield',
+      '#title' => t('Simple Search Label'),
+      '#size' => 30,
+      '#default_value' => variable_get('islandora_solr_simple_search_label_title', 'Search Term'),
+      '#description' => t('Customize label text associated with search input form. Accessibility guidelines recommend whenever possible to explicitly associate label text with form inputs.'),
```
Comment (whikloj) [inline includes/blocks.inc:119]: @dmcollins Just wondering about the "Accessibility guidelines...." note. The form requires you to enter a label and the generated form properly associates the label as a `<label for="simple-search-textfield">`. So while informative I'm not clear on its purpose. What are you trying to accomplish with that text?

Comment (dmcollins) [reply includes/blocks.inc:119]: @whikloj Mostly it's there to explain why the label is required in the first place combined with a checkbox (rather than just having a blank label, for example).

**Comment by whikloj** [comment]
@dmcollins I merged another PR in islandora_solr_search, so now you have a conflict. Can you rebase this branch? Ping me if you need help.

**Comment by dmcollins** [comment]
@whikloj I couldn't quite figure out how to rebase this without fear of messing up everything; but I do think I isolated the conflicts and corrected them...

**Comment by whikloj** [comment]
@dmcollins there still seems to be a conflict. 

Here is what I do to rebase a branch. 

1. Change into the directory for the module.
1. Checkout your copy of the master branch (this is what you are merging against so origin/7.x in this case). 
1. git pull all updates to the master branch.
1. Checkout your PR branch (dmcollins/ISLANDORA-1743 in this case)
1. rebase as `git rebase <master branch name>`

See this page about [rebase conflicts](https://help.github.com/articles/resolving-merge-conflicts-after-a-git-rebase/) (a rebase is just a fancy merge) and this page about [merge conflicts](https://help.github.com/articles/resolving-a-merge-conflict-using-the-command-line/).

TL;DR is edit the file with conflicts file, and fix it. Then `git add` it and run `git rebase --continue`. If it is totally messed up, run `git rebase --abort` to go back to the start. You can always recover.

Here are the steps I would do in my structure where 7.x is the master branch and ISLANDORA-1743 is your PR branch

```
cd islandora_solr_search
git checkout 7.x
git pull
git checkout ISLANDORA-1743
git rebase 7.x
... probably fix islandora_solr.install
git add islandora_solr.install
git rebase --continue
```

Because Git rebase replays each commit on the end of the 7.x branch you might have to fix the islandora_solr.install file more than once. You only have 4 commits so you should be good.

**Comment by whikloj** [comment]
Sorry I should add, that if your rebase is successful. You will need to **force** push your branch back to your repo. This is because a rebase alters all the checksums, so the repository sees different commits on each branch.

You'll get a message like _The remote repo has 4 commits and the local repo has 5 commits each...._

I'd just do `git push -f dmcollins ISLANDORA-1743` or whatever your local remote and branch are called.

**Comment by dmcollins** [comment]
@whikloj Alright! I think I've done it. As of writing this comment Travis is running so I guess we will find out.

**Comment by dmcollins** [comment]
And it passed! Thanks @whikloj for walking me through the rebase!

**Review by DiegoPino — APPROVED** [review]
Code looks good. Do we need screenshot and new doc for the upcoming 7.x-1.10 release? @Islandora/7-x-1-x-committers

**Comment by whikloj** [comment]
@DiegoPino Solr is one of those modules with so many options that we don't seem to document any of them. This is a simple block level configuration and I think it is small enough that the information on the screen should be good. Unless @dmcollins wants to add a note to the README explaining that you can now configure the label in the block config?

**Comment by DiegoPino** [comment]
@whikloj i agree. Just scared of the amount of toggle and switches we have that are undocumented.

**Comment by dmcollins** [comment]
I'll update the README now

**Review by whikloj — APPROVED** [review]

**Comment by DiegoPino** [comment]
@dmcollins++ looks ready to be merged. Will leave the actual reviewer the honors. good job
